From c52f196d510093083e527c93278c8a3add286b36 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Tue, 18 Jul 2017 14:54:43 +0800
Subject: [PATCH 2150/5242] MLK-16015-1 phy: mixel-lvds: Add delay to wait PHY
 to be locked

commit  356465fdde5b13d3f5533c3894c37572f2db3fcf from
https://source.codeaurora.org/external/imx/linux-imx.git

It seems that we haven't got SCU ISO bit available to check if PHY is
locked or not after enable, so let's simply delay for a while as a
temporary solution.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/phy/phy-mixel-lvds.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/phy/phy-mixel-lvds.c b/drivers/phy/phy-mixel-lvds.c
index 01ac237..824b251 100644
--- a/drivers/phy/phy-mixel-lvds.c
+++ b/drivers/phy/phy-mixel-lvds.c
@@ -13,6 +13,7 @@
  */
 
 #include <linux/clk.h>
+#include <linux/delay.h>
 #include <linux/io.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
@@ -166,6 +167,8 @@ static int mixel_lvds_phy_power_on(struct phy *phy)
 	phy_write(phy, CH_EN(id), PHY_CTRL + SET);
 	mutex_unlock(&priv->lock);
 
+	usleep_range(500, 1000);
+
 	return 0;
 }
 
-- 
1.7.9.5

